float64 mx
float64 my
